### DISCUSSION TITLE
Cannot find module 'xstate' or its corresponding type declarations

### DISCUSSION THREAD

**Discussion by lanmonster** [statelyai/xstate#1519] [Q&A] [answered]
I forked the repo, cloned, ran yarn, then tried to run the tests and got an error that xstate could not be found. It seems to find it ok in the production code, but the tests do not work.

```
error TS2307: Cannot find module 'xstate' or its corresponding type declarations.

    3 import { createMachine, sendParent, Actor, assign, spawn } from 'xstate';
```

David mentioned that @Andarist might be able to help.

Thanks!

**Comment by Andarist** [comment, marked as answer]
My preferred way to work with the codebase is to run `yarn dev` in the root. This builds all packages first and then starts a test with a watcher. Alternatively, you can run `yarn build` in the root and then pick a package in which you'd like to run tests. 

`yarn dev` ensures that all packages are constantly re-built though, so you won't end up with weird situations that a dependant package is being tested against outdated dependency code even though you have made changes to source files of the particular dependency.
